public/src/components/teacher/Results.tsx

```diff
@@ -134,21 +134,8 @@ export class Results extends React.Component<IResultsProp, IResultsState> {
         });
     }
 
-    private handleOnchange(query: string): void {
-        query = query.toLowerCase();
-        const filteredData: IStudentLabsForCourse[] = [];
-        this.props.students.forEach((std) => {
-            const usr = std.enrollment.getUser();
-            if (usr) {
-                if (usr.getName().toLowerCase().indexOf(query) !== -1
-                    || usr.getEmail().toLowerCase().indexOf(query) !== -1
-                    || usr.getLogin().toLowerCase().indexOf(query) !== -1
-                ) {
-                    filteredData.push(std);
-                }
-            }
-        });
-
+    private handleSearch(query: string): void {
+        const filteredData = searchForLabs(this.props.students, query);
```

Comment:
Can be called inline in setState call below

Reply:
Done.